### PR TITLE
add scikit-sparse

### PR DESCRIPTION
**Pull request by grlee77**
This python package wraps the CHOLMOD functionality from SuiteSparse and is complementary to scikit-umfpack (#683).

Note that the wrapper code here is BSD 2-clause, but the underlying CHOLMOD library from SuiteSparse is LGPL v2.1 which is perhaps the more relevant constraint for other projects wanting to use this.  Should that be noted in the license section? 

also, any other volunteers for maintaining?

### CODE REVIEW

**Comment by grlee77** [comment]
apologies for forgetting [skip appveyor].  feel free to cancel that one.

**Comment by grlee77** [comment]
looks like one test fails on an import from `pkg_resources` which I believe is part of `setuptools`?  I will try adding setuptools to the test requirements and see if that helps

**Comment by jakirkham** [comment]
Yes, glad to see this working. 😄 Basically, if you need `setuptools`, it needs to be explicitly included in your build requirements as it will not be included otherwise.

**Comment by jakirkham** [comment]
So, the problem is `bob.ip.flandmark`, which is awaiting feedstocking. Could you please try merging with/rebasing on `master` and removing anything that is not `scikit-sparse`.

**Comment by grlee77** [comment]
rebased, but forgot to skip appveyor yet again :(

**Comment by mariusvniekerk** [comment]
you probably need to have setuptools in your run dependency section as well

**Comment by grlee77** [comment]
All green!  Thanks @mariusvniekerk
also updated the license to LGPL 2.1 to reflect the license of the underlying CHOLMOD library

**Comment by jakirkham** [comment]
Are we statically linking to CHOLMOD, @grlee77?

**Comment by grlee77** [comment]
It is dynamically linked to libcholmod, so I agree it is probably be okay to list as BSD 2-clause.  will go ahead and rebase without the last commit

**Comment by grlee77** [comment]
I think this one is ready to go

File: recipes/scikit-sparse/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - numpy x.x
+    - suitesparse
+    - cython
+    - setuptools
+  run:
+    - python
+    - numpy x.x
+    - scipy
+    - suitesparse
+    - setuptools
```
Comment (grlee77) [inline recipes/scikit-sparse/meta.yaml]: I don't think it should be, but the tests wouldn't pass without it.  Looking into it, there is one place in the tests that does:
`from pkg_resources import resource_filename`
I was getting an ImportError on pkg_resources.  I will try moving it under test/requires instead and see how that goes.

**Comment by grlee77** [comment]
@jakirkham 
`setuptools` has been moved to the test requirements instead

**Comment by jakirkham** [comment]
Just adding this to the record so we have it to go back to.

One of the tests from `scikit-sparse` expects to be able to import `pkg_resources`. The failure one gets without this is shown below.

```
======================================================================
ERROR: sksparse.test_cholmod.test_cholesky_matrix_market
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/zopt/conda2/envs/_test/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/zopt/conda2/envs/_test/lib/python2.7/site-packages/sksparse/test_cholmod.py", line 164, in test_cholesky_matrix_market
    X = mm_matrix(problem)
  File "/zopt/conda2/envs/_test/lib/python2.7/site-packages/sksparse/test_cholmod.py", line 155, in mm_matrix
    from pkg_resources import resource_filename
ImportError: No module named pkg_resources

----------------------------------------------------------------------
```

File: recipes/scikit-sparse/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/scikit-sparse/meta.yaml]: I see. I think this issue was encountered when `bob.ip.flandmark` was around. So assumed it was related to that.

Honestly, I can see this both ways. Though I think you are right to move it to `test` requirements. We did the same thing for `nose` with `numpy`.

File: recipes/scikit-sparse/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - nosetests -sv sksparse
+
+about:
+  home: https://github.com/scikit-sparse/scikit-sparse
+  license: BSD 2-clause
+  summary: A companion to the scipy.sparse library for sparse matrix manipulation in Python.
+
+extra:
+    recipe-maintainers:
+      - grlee77
```
Comment (jakirkham) [inline recipes/scikit-sparse/meta.yaml]: Spacing is a little funky here, but it is not a blocker. Feel free to address this later.

**Comment by jakirkham** [comment]
Thanks @grlee77.

**Pull request merged**